src/main/kotlin/io/gitlab/arturbosch/detekt/idea/DetektAnnotator.kt

```diff
@@ -22,7 +22,7 @@ class DetektAnnotator : ExternalAnnotator<PsiFile, List<Finding>>() {
         if (!collectedInfo.project.isDetektEnabled()) {
             return emptyList()
         }
-        if (collectedInfo.isKotlinFile()) {
```

Comment:
This line was wrong. It would have skipped every Kotlin file but did not because apparently `psiFile.language != KotlinLanguage.INSTANCE`,
Testing for `KtFile` does not work either as IntelliJ may use some kind of LightPsiElement structures.
We have to compare language id's here. Kotlin light elements also use KotlinLanguage.INSTANCE.